src/modules/twap/updaters/TwapOrdersUpdater.tsx

```diff
@@ -0,0 +1,72 @@
+import { useUpdateAtom } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+import { ComposableCoW } from 'abis/types'
+
+import { useDiscreteOrdersFromOrderBook } from '../hooks/useDiscreteOrdersFromOrderBook'
+import { useFetchDiscreteOrders } from '../hooks/useFetchDiscreteOrders'
+import { useFetchTwapOrdersFromSafe } from '../hooks/useFetchTwapOrdersFromSafe'
+import { useTwapOrdersAuthMulticall } from '../hooks/useTwapOrdersAuthMulticall'
+import { twapDiscreteOrdersAtom } from '../state/twapDiscreteOrdersAtom'
+import { twapOrdersListAtom } from '../state/twapOrdersListAtom'
+import { TwapOrderInfo } from '../types'
+import { getConditionalOrderId } from '../utils/getConditionalOrderId'
+import { getTwapOrdersItems } from '../utils/getTwapOrdersItems'
+import { isTwapOrderExpired } from '../utils/getTwapOrderStatus'
+import { parseTwapOrderStruct } from '../utils/parseTwapOrderStruct'
+
+export function TwapOrdersUpdater(props: {
+  safeAddress: string
+  chainId: SupportedChainId
+  composableCowContract: ComposableCoW
```

Comment:
why making the updater an hybrid (receiving parameters and using some hooks to get them too).
Shouldn't it have no params?